### PR TITLE
Add pxetools for easy netboot server. Improve netboot documentation.

### PR DESCRIPTION
**Pull request by liamfraser**

### CODE REVIEW

**Comment by lurch** [comment]
According to the [Travis report](https://travis-ci.org/raspberrypi/documentation/builds/147989932) `hardware/raspberrypi/bootmodes/net_tutorial.md` still links to `bootcode.bin` and `start.elf`

File: hardware/raspberrypi/bootmodes/pxetools/prepare_pxetools [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+cd /nfs/bases
+sudo wget -O raspbian_latest.zip https://downloads.raspberrypi.org/raspbian_lite_latest
+sudo unzip raspbian_latest.zip
+sudo rm raspbian_latest.zip
+
+sudo wget  -O /usr/local/sbin/pxetools https://raw.githubusercontent.com/raspberrypi/documentation/master/hardware/raspberrypi/bootmodes/pxetools/pxetools
+sudo chmod +x /usr/local/sbin/pxetools
+
+sudo sed -i "s,LAN = \"10.3.14.0/24\",LAN = \"$NETWORK\"," /usr/local/sbin/pxetools
+sudo sed -i "s,NFS_IP = \"10.3.14.18\",NFS_IP = \"$IP\"," /usr/local/sbin/pxetools
```
Comment (lurch) [inline hardware/raspberrypi/bootmodes/pxetools/prepare_pxetools]: Why do these values get hard-coded? Wouldn't it be better for the `pxetools` script to determine them dynamically, the same way this script does?

File: hardware/raspberrypi/bootmodes/pxetools/pxetools [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    name = input("Name for pi: ")
+    
+    print("Select a base image:")
+    selection = ["I will prepare my own filesystem"] + os.listdir('/nfs/bases')
+    for i in range (0, len(selection)):
+        print("\t{}. {}".format(i + 1, selection[i]))
+    
+    img_choice = input("Enter an option number: ")
+
+    # Validate stuff
```
Comment (lurch) [inline hardware/raspberrypi/bootmodes/pxetools/pxetools]: All this validation of the `serial` value should probably be done **before** asking the user for input?

File: hardware/raspberrypi/bootmodes/netboot_server_easy.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+## pxetools
+We have created a Python script that is used internally to quickly set up Pi's that will network boot. It takes a serial number, which you can find in `cat /proc/cpuinfo`, an owner name and the name of the Pi. It then creates a root filesystem for that Pi from a raspbian image. There is also a --list option which will print out the IP address of the Pi, and remove an option. The follwing instructions describe how to set up the environment required by the script starting from a fresh Raspbian lite image. It might be a good idea to mount a hard disk or flash drive on /nfs so that your SD card isn't providing filesystems to multiple Pi's. This is left as an exercise for the reader.
```
Comment (lurch) [inline hardware/raspberrypi/bootmodes/netboot_server_easy.md]: raspbian should probably have a capital R.
Was "and remove an option" supposed to be "and a --remove option" ?
"Pi's" shouldn't have an apostrophe.

**Comment by lurch** [comment]
I wonder if the `prepare_pxetools` and `pxetools` scripts should be moved to the https://github.com/raspberrypi/tools repo, leaving https://github.com/raspberrypi/documentation purely for documentation?

**Comment by liamfraser** [comment]
Will do these changes this today. Going to keep the scripts in documentation though, having multiple repositories makes it more of a pain to maintain. Thanks.

**Pull request merged**